src/devnet-utils.ts

```diff
@@ -56,94 +57,125 @@ export class DevnetUtils implements Devnet {
         return `${this.hre.starknet.networkConfig.url}`;
     }
 
-    private async withErrorHandler<T>(asyncFn: () => Promise<T>, errorMessage: string) {
+    private async requestHandler(
+        url: string,
+        method: Method,
+        errorMessage?: string,
+        data?: unknown,
+        headers?: any
+    ): Promise<AxiosResponse> {
         try {
-            return await asyncFn();
+            // Make the request
+            const axiosInstance = axios.create({
+                baseURL: this.endpoint,
+                timeout: REQUEST_TIMEOUT,
+                timeoutErrorMessage: "Request timed out"
+            });
+
+            return axiosInstance.request({
+                url,
+                method,
+                data,
+                headers
+            });
         } catch (error) {
             const parent = error instanceof Error && error;
-
             throw new StarknetPluginError(errorMessage, parent);
         }
     }
 
     public async restart() {
-        return this.withErrorHandler<void>(async () => {
-            await axios.post(`${this.endpoint}/restart`);
-        }, "Failed to restart the devnet!");
+        await this.requestHandler("/restart", "POST", "Failed to restart the devnet!");
     }
 
     public async flush() {
-        return this.withErrorHandler<FlushResponse>(async () => {
-            const response = await axios.post<FlushResponse>(`${this.endpoint}/postman/flush`);
-            return response.data;
-        }, "Request failed. Make sure your network has the /postman endpoint");
+        const response = await this.requestHandler(
+            "/postman/flush",
+            "POST",
+            "Request failed. Make sure your network has the /flush endpoint"
+        );
+        return response.data as FlushResponse;
```

Comment:
Instead of casting here, could `requestHandler` be parametrized with a generic `T` (kind of like it used to be)

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/274/commits/cad87ed19051704fb1e0dc240df16f8583ab878e